src/basic/fd-util.c

```diff
@@ -476,6 +479,17 @@ int fd_get_path(int fd, char **ret) {
         return r;
 }
 
+int fd_is_on_btrfs(int fd) {
+        struct statfs sfs;
+
+        assert(fd >= 0);
+
+        if (fstatfs(fd, &sfs) < 0)
+                return -errno;
+
+        return F_TYPE_EQUAL(sfs.f_type, BTRFS_SUPER_MAGIC);
+}
```

Comment:
drop the function altogether.  `fd_is_fs_type(fd, BTRFS_SUPER_MAGIC)` does the same and is not much longer to type